docs/user-guide/persistent-volumes/index.md

```diff
@@ -399,3 +399,27 @@ parameters:
 
 * `type`: [VolumeType](http://docs.openstack.org/admin-guide/dashboard-manage-volumes.html) created in Cinder. Default is empty.
 * `availability`: Availability Zone. Default is empty.
+
+## Writing Portable Configuration
+
+When writing configuration templates or examples that need persistent storage
+and which are intended to run on a wide range of clusters, we recommend the following pattern:
+
+- Do include PersistentVolumeClaim objects in your bundle of config (alongside Deployments, ConfigMaps, etc).
+- Do not include PersistentVolume objects in the config, since the user instantiating the config may not have
+  permission to create PersistentVolumes.
+- Give the user the option of providing a storage class name when instantating the template.
+- If the user provides a storage class name, and the cluster is version 1.4 or newer, put that value into the `volume.beta.kubernetes.io/storage-class` annotation of the PVC.
```

Comment:
These next two bullets, related to storage class names, should be sub-bullets of the previous one (line 411).